contracts/root/StakeManager.sol

```diff
@@ -79,31 +79,39 @@ contract StakeManager is StakeManagerInterface, RootChainable, Lockable {
   }
   
   function stake(uint256 amount, bytes data) public {
-    require(_validatorThreshold > nextValidatorList.currentSize(), "Next validator set full");
-    require(amount < MAX_UINT96, "Stay realistic!!");
-    require(stakers[msg.sender].epoch == 0, "No second time staking");
     stakeFor(msg.sender, amount, data);
   }
 
-  // function nextValidatorSetSize() public view  returns(uint256) {
-  //   return nextValidatorList.currentSize();
-  // }
+  function validatorSetSize() public view  returns(uint256) {
+    return validatorSetSize;
+  }
 
-  function getDetails(address user) public view returns(uint256 , uint256, ValidatorStatus) {
-    return (stakers[user].activationEpoch, stakers[user].deActivationEpoch, stakers[user].status);
+  function nextValidatorSetSize() public view  returns(uint256) {
+    return nextValidatorList.currentSize();
+  }
+
+  function getDetails(address user) public view returns(uint256 , uint256) {
+    return (stakers[user].activationEpoch, stakers[user].deActivationEpoch);
+  }
+
+  function currentValidatorsTotalStake(uint256 epoch) public view returns(uint256) {
```

Comment:
No need to create method for it. public mapping will expose public method for the same automatically. Like `web3Contract.totalValidatorStake(epoch)`